components/Modals/Modal.tsx

```diff
@@ -0,0 +1,86 @@
+import styled, { ThemeContext } from 'styled-components'
+import { useSpring, animated } from 'react-spring'
+import { MdClose } from 'react-icons/md'
+import { ModalProps } from '../../lib/gymmieInterfaces'
+
+export default function Modal({
+  modalIsOpen,
+  modalType,
+  modalText,
+  closeModal,
+}: ModalProps) {
+  const animation = useSpring({
+    config: {
+      duration: 250,
+    },
+    opacity: modalIsOpen ? 1 : 0,
+    transform: modalIsOpen ? `translateY(0%)` : `translateY(-100%)`,
+  })
+
+  return (
+    <>
+      {modalIsOpen ? (
+        <Background type={modalType}>
+          <animated.div style={animation}>
+            <ModalWrapper>
+              <ModalContent>{modalText}</ModalContent>
+              <CloseModalButton aria-label="Close modal" onClick={closeModal} />
+            </ModalWrapper>
+          </animated.div>
+        </Background>
+      ) : null}
+    </>
+  )
+}
+
+const Background = styled.div<ModalProps>`
```

Comment:
Pick out only the type you're using here, so:

```javascript
const Background = styled.div<Pick<ModalProps, 'modalType'>>``

```
Like this; you're styled component typing is more specific (and therefor safer)